### PR TITLE
ukmm: init at 0.7.1

### LINKED ISSUE #225486: Package request: python3*Packages.BCML
**Project description**
A cross platform mod loader for The Legend of Zelda: Breath of the Wild in CEMU.

**Metadata**

* homepage URL: https://github.com/NiceneNerd/BCML
* source URL: https://github.com/NiceneNerd/BCML
* license: GPL3+
* platforms: Windows, Linux

### PR DESCRIPTION
**Pull request by lasers**
###### Description of changes

A new mod manager for The Legend of Zelda: Breath of the Wild. Successor to BCML. 
source & homepage URL: https://github.com/NiceneNerd/ukmm/

Closes https://github.com/NixOS/nixpkgs/issues/225486.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/ukmm/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    glib
+    gtk3-x11
+  ];
+
+  meta = with lib; {
+    description = "A new mod manager for The Legend of Zelda: Breath of the Wild. Successor to BCML.";
+    homepage = "https://github.com/NiceneNerd/ukmm";
+    changelog = "https://github.com/NiceneNerd/ukmm/releases/tag/v${version}";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [];
```
Comment (kira-bruneau) [inline pkgs/applications/emulators/ukmm/default.nix:57]: This shouldn't be empty (See [CONTRIBUTING.md#submitting-changes](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes)), would you be willing to maintain this?

You could add yourself to [the maintainers list](https://github.com/NixOS/nixpkgs/blob/f17aa727523ece00fcadf56f49340bcdd81a74c1/maintainers/maintainer-list.nix).

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 226992` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ukmm</li>
  </ul>
</details>

**Review by kira-bruneau — COMMENTED** [review]
Thanks @lasers!! This looks a lot easier to maintain than BCML :sweat_smile:.

I just noticed one thing:

File: pkgs/applications/emulators/ukmm/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  ];
+
+  buildInputs = [
+    atk
+    glib
+    gtk3-x11
+  ];
+
+  meta = with lib; {
+    description = "A new mod manager for The Legend of Zelda: Breath of the Wild. Successor to BCML.";
```
Comment (kira-bruneau) [inline pkgs/applications/emulators/ukmm/default.nix:53]: Oh also from [CONTRIBUTING.md#submitting-changes](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes), this shouldn't end in a period.

File: pkgs/applications/emulators/ukmm/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        "egui-notify-0.4.0" = "sha256-jybtUnv9xqzulZ5nfg+T1u8iTOsPjKGVVQ7JhwbvPdU=";
+        "egui_commonmark-0.6.0" = "sha256-hsVbtL2F+jifnzN6FgcDAVtLd1bVxTs0twn0SMvq9eU=";
+        "egui_dock-0.2.1" = "sha256-gGIO0boXKxLu0ABDH/uJhEZEoE/ql8E65LRmr0Xhv3s=";
+        "junction-0.2.0" = "sha256-6+pPp5wG1NoIj16Z+OvO4Pvy0jnQibn/A9cTaHAEVq4=";
+        "msbt-0.1.1" = "sha256-PtBs60xgYrwS7yPnRzXpExwYUD3azIaqObRnnJEL5dE=";
+        "msyt-1.2.1" = "sha256-aw5whCoQBhO0u9Fx2rTO1sRuPdGnAAlmPWv5q8CbQcI=";
+      };
+  };
+
+  doCheck = false;
```
Comment (kira-bruneau) [inline pkgs/applications/emulators/ukmm/default.nix:37]: I'm also curious - what was wrong with the tests?

File: pkgs/applications/emulators/ukmm/default.nix (hunk 2 above)
Comment (lasers) [reply pkgs/applications/emulators/ukmm/default.nix:53]: Done.
```diff
- A new mod manager for The Legend of Zelda: Breath of the Wild. Successor to BCML.
+ A mod manager for The Legend of Zelda: Breath of the Wild
```
Will push out later after more reviews/questions.

File: pkgs/applications/emulators/ukmm/default.nix (hunk 3 above)
Comment (lasers) [reply pkgs/applications/emulators/ukmm/default.nix:37]:
```
...
...
...
     Compiling ukmm v0.7.1 (/build/source)
      Finished release [optimized] target(s) in 1m 21s
       Running unittests src/main.rs (target/x86_64-unknown-linux-gnu/release/deps/ukmm-681b477a3355b37b)
   
  running 1 test
  test gui::tasks::tests::remerge ... FAILED
   
  failures:
   
  ---- gui::tasks::tests::remerge stdout ----
  thread 'gui::tasks::tests::remerge' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to initialize mod manager at `crates/uk-manager/src/core.rs@21:44`
   
  Caused by:
      0: failed to create directory `/homeless-shelter/.local/share/ukmm/wiiu/profiles/Default`
      1: Permission denied (os error 13)', src/gui/tasks.rs:541:54
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
   
   
  failures:
      gui::tasks::tests::remerge
   
  test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
   
  error: test failed, to rerun pass `--bin ukmm`
```

File: pkgs/applications/emulators/ukmm/default.nix (hunk 1 above)
Comment (lasers) [reply pkgs/applications/emulators/ukmm/default.nix:57]: I haven't played any Zelda in my life so I don't think I want to pay attention to this. Also, I don't know nix language. Just borrowing codes from others and piercing puzzles together enough to get this working.

File: pkgs/applications/emulators/ukmm/default.nix (hunk 3 above)
Comment (kira-bruneau) [reply pkgs/applications/emulators/ukmm/default.nix:37]: Oh, I think you can fix that with:

```nix
preCheck = ''
  export HOME=$TMDIR
'';
```

File: pkgs/applications/emulators/ukmm/default.nix (hunk 1 above)
Comment (kira-bruneau) [reply pkgs/applications/emulators/ukmm/default.nix:57]: Ok - I'd be happy to maintain it then!

File: pkgs/applications/emulators/ukmm/default.nix (hunk 3 above)
Comment (lasers) [reply pkgs/applications/emulators/ukmm/default.nix:37]:
```diff
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';```
```
```
     Compiling ukmm v0.7.1 (/build/source)
      Finished release [optimized] target(s) in 1m 15s
       Running unittests src/main.rs (target/x86_64-unknown-linux-gnu/release/deps/ukmm-681b477a3355b37b)
   
  running 1 test
  test gui::tasks::tests::remerge ... FAILED
   
  failures:
   
  ---- gui::tasks::tests::remerge stdout ----
  thread 'gui::tasks::tests::remerge' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to apply pending mod changes
   
  Caused by:
      No dump available for current platform at `crates/uk-manager/src/deploy.rs@443:14`', src/gui/tasks.rs:542:51
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
   
   
  failures:
      gui::tasks::tests::remerge
   
  test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
   
  error: test failed, to rerun pass `--bin ukmm`
```

**Comment by kira-bruneau** [comment]
I just added myself as a maintainer & finished up enabling the tests

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 226992` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ukmm</li>
  </ul>
</details>

**Pull request merged**